src/GoogleFit/FitOnFhir.GoogleFit.Tests/GoogleFitImportServiceTests.cs

```diff
@@ -0,0 +1,221 @@
+
+        private readonly GoogleFitUser _googleFitUser;
+        private readonly MockEventHubProducerClient _eventHubProducerClient;
+        private readonly MockFaultyEventHubProducerClient _faultyEventHubProducerClient;
+        private readonly EventData _eventData;
+        private readonly MedTechDataset _medTechDataset;
+
+        private readonly IGoogleFitClient _googleFitClient;
+        private readonly GoogleFitImportOptions _options;
+        private readonly Func<DateTimeOffset> _utcNowFunc;
+        private readonly MockLogger<GoogleFitImportService> _importServiceLogger;
+        private readonly ITelemetryLogger _telemetryLogger;
+        private IGoogleFitImportService _googleFitImportService;
+
+        public GoogleFitImportServiceTests()
+        {
+            // _googleFitUser = new GoogleFitUser(_googleUserId);
+            _googleFitUser = Substitute.For<GoogleFitUser>(_googleUserId);
+            _eventData = new EventData(_eventBody);
+            _medTechDataset = new MedTechDataset(_dataset, _dataSource);
+            _dataSources.Add(_dataSource);
+            _faultyEventHubProducerClient = new MockFaultyEventHubProducerClient();
+
+            // GoogleFitImportService dependencies
+            _googleFitClient = Substitute.For<IGoogleFitClient>();
+            _eventHubProducerClient = new MockEventHubProducerClient();
+            _options = Substitute.For<GoogleFitImportOptions>();
+            _utcNowFunc = Substitute.For<Func<DateTimeOffset>>();
+            _now = DateTimeOffset.Parse("01/12/2004");
+            _utcNowFunc().Returns(_now);
+            _importServiceLogger = Substitute.For<MockLogger<GoogleFitImportService>>();
+            _telemetryLogger = Substitute.For<ITelemetryLogger>();
+
+            // create the service
+            _googleFitImportService = new GoogleFitImportService(
+                _googleFitClient,
+                _eventHubProducerClient,
+                _options,
+                _utcNowFunc,
+                _importServiceLogger,
+                _telemetryLogger);
+        }
+
+        [Fact]
+        public async Task GivenNoLastSyncStored_WhenTryGetLastSyncTime_GeneratesDataSetIdForLast30Days()
+        {
+            string thirtyDaysBackDataSetId = "1071291600000000000-1073883600000000000";
+
+            await _googleFitImportService.ProcessDatasetRequests(_googleFitUser, _dataSources, _tokensResponse, _cancellationToken);
+
+            _ = _googleFitClient.Received(1).DatasetRequest(
+                Arg.Is<string>(access => access == _tokensResponse.AccessToken),
+                Arg.Is<DataSource>(ds => ds == _dataSource),
+                Arg.Is<string>(str => str == thirtyDaysBackDataSetId),
+                Arg.Is<CancellationToken>(token => token == _cancellationToken));
+        }
+
+        [Fact]
+        public async Task GivenLastSyncStored_WhenTryGetLastSyncTime_GeneratesDataSetIdForLastSync()
```

Comment:
This test fails on my machine with the following message:

`NSubstitute.Exceptions.ReceivedCallsException : Expected to receive exactly 1 call matching:
	DatasetRequest(access => (access == value(FitOnFhir.GoogleFit.Tests.GoogleFitImportServiceTests)._tokensResponse.AccessToken), ds => (ds == value(FitOnFhir.GoogleFit.Tests.GoogleFitImportServiceTests)._dataSource), str => (str == value(FitOnFhir.GoogleFit.Tests.GoogleFitImportServiceTests+<>c__DisplayClass27_0).oneDayBackDataSetId), token => (token == value(FitOnFhir.GoogleFit.Tests.GoogleFitImportServiceTests)._cancellationToken), <null>)
Actually received no matching calls.
Received 1 non-matching call (non-matching arguments indicated with '*' characters):
	DatasetRequest("AccessToken", DataSource, *"1073808000000000000-1073894400000000000"*, System.Threading.CancellationToken, <null>)`